diff --git a/node_modules/react-native-screens/lib/commonjs/components/Screen.js b/node_modules/react-native-screens/lib/commonjs/components/Screen.js
index db0f2e3..084ad2c 100644
--- a/node_modules/react-native-screens/lib/commonjs/components/Screen.js
+++ b/node_modules/react-native-screens/lib/commonjs/components/Screen.js
@@ -80,15 +80,18 @@ const InnerScreen = exports.InnerScreen = /*#__PURE__*/_react.default.forwardRef
         setRef(ref);
       }
     };
+    const sheetProps = stackPresentation === 'formSheet' ? {
+      sheetAllowedDetents,
+      sheetLargestUndimmedDetent,
+      sheetGrabberVisible,
+      sheetCornerRadius,
+      sheetExpandsWhenScrolledToEdge,
+    } : {};
     return /*#__PURE__*/_react.default.createElement(_DelayedFreeze.default, {
       freeze: freezeOnBlur && activityState === 0
     }, /*#__PURE__*/_react.default.createElement(AnimatedScreen, _extends({}, props, {
       activityState: activityState,
-      sheetAllowedDetents: sheetAllowedDetents,
-      sheetLargestUndimmedDetent: sheetLargestUndimmedDetent,
-      sheetGrabberVisible: sheetGrabberVisible,
-      sheetCornerRadius: sheetCornerRadius,
-      sheetExpandsWhenScrolledToEdge: sheetExpandsWhenScrolledToEdge,
+    }, sheetProps, {
       gestureResponseDistance: {
         start: gestureResponseDistance?.start ?? -1,
         end: gestureResponseDistance?.end ?? -1,
diff --git a/node_modules/react-native-screens/lib/module/components/Screen.js b/node_modules/react-native-screens/lib/module/components/Screen.js
index ec8f973..ef55799 100644
--- a/node_modules/react-native-screens/lib/module/components/Screen.js
+++ b/node_modules/react-native-screens/lib/module/components/Screen.js
@@ -75,15 +75,18 @@ export const InnerScreen = /*#__PURE__*/React.forwardRef(function InnerScreen(pr
         setRef(ref);
       }
     };
+    const sheetProps = stackPresentation === 'formSheet' ? {
+      sheetAllowedDetents,
+      sheetLargestUndimmedDetent,
+      sheetGrabberVisible,
+      sheetCornerRadius,
+      sheetExpandsWhenScrolledToEdge,
+    } : {};
     return /*#__PURE__*/React.createElement(DelayedFreeze, {
       freeze: freezeOnBlur && activityState === 0
     }, /*#__PURE__*/React.createElement(AnimatedScreen, _extends({}, props, {
       activityState: activityState,
-      sheetAllowedDetents: sheetAllowedDetents,
-      sheetLargestUndimmedDetent: sheetLargestUndimmedDetent,
-      sheetGrabberVisible: sheetGrabberVisible,
-      sheetCornerRadius: sheetCornerRadius,
-      sheetExpandsWhenScrolledToEdge: sheetExpandsWhenScrolledToEdge,
+    }, sheetProps, {
       gestureResponseDistance: {
         start: gestureResponseDistance?.start ?? -1,
         end: gestureResponseDistance?.end ?? -1,
